libs/core/src/schemas/community.schema.ts

```diff
@@ -0,0 +1,79 @@
+import { z } from 'zod';
+import { ChainBase, ChainNetwork, ChainType, DefaultPage } from '../types';
+
+export const CommunityStake = z.object({
+  id: z.number().optional(),
+  community_id: z.string().optional(),
+  stake_id: z.number().optional(),
+  stake_token: z.string().optional(),
+  vote_weight: z.number().optional(),
+  stake_enabled: z.boolean().optional(),
+  created_at: z.date().optional(),
+  updated_at: z.date().optional(),
+});
+
+export const Community = z.object({
+  name: z.string(),
+  chain_node_id: z.number(),
+  default_symbol: z.string(),
+  network: z.nativeEnum(ChainNetwork),
+  base: z.nativeEnum(ChainBase),
+  icon_url: z.string(),
+  active: z.boolean(),
+  type: z.nativeEnum(ChainType),
+  id: z.string().optional(),
+  description: z.string().optional(),
+  social_links: z.array(z.string()).optional(),
+  ss58_prefix: z.number().optional(),
+  stages_enabled: z.boolean().optional(),
+  custom_stages: z.array(z.string()).optional(),
+  custom_domain: z.string().optional(),
+  block_explorer_ids: z.string().optional(),
+  collapsed_on_homepage: z.boolean().optional(),
+  substrate_spec: z.string().optional(),
+  has_chain_events_listener: z.boolean().optional(),
+  default_summary_view: z.boolean().optional(),
+  default_page: z.nativeEnum(DefaultPage).optional(),
+  has_homepage: z.boolean().optional(),
+  terms: z.string().optional(),
+  admin_only_polling: z.boolean().optional(),
+  bech32_prefix: z.string().optional(),
+  hide_projects: z.boolean().optional(),
+  token_name: z.string().optional(),
+  ce_verbose: z.boolean().optional(),
+  discord_config_id: z.number().optional(),
+  category: z.unknown().optional(), // Assuming category can be any type
+  discord_bot_webhooks_enabled: z.boolean().optional(),
+  directory_page_enabled: z.boolean().optional(),
+  directory_page_chain_node_id: z.number().optional(),
+  namespace: z.string().optional(),
+  redirect: z.string().optional(),
+  created_at: z.date().optional(),
+  updated_at: z.date().optional(),
+});
+
+export const GetCommunityStakeSchema = {
```

Comment:
Schema suffix might be redundant since this belongs to the schemas namespace... guess I'm trying to find the minimal convention 😬

Reply:
should we have optional stake in community?

Reply:
I don't think we should have optional stake in community.

These Community/CommunityStake schemas are representations of the database model. I had re-wrote them in zod because:
1. single-source of truth: If we use them as outputs, then its good to have a zod schema, which generates the db model types
2. Ryan is working on the seed data, which could take in these schemas to create data for random tables, which he can insert into the database.

So for these reasons we should not treat these as schemas of the aggregate, but instead of schemas for the db tables.

Reply:
Let's discuss in our next meeting... I'm worried about commands that return partial aggregate states, the stake is part of the logical aggregate... db schema concerns shouldn't leak the core model.